src/couch/src/couch_att.erl

```diff
@@ -680,11 +680,24 @@ is_compressible(Type) ->
 
 
 max_attachment_size() ->
-    case config:get("couchdb", "max_attachment_size", "infinity") of
+    max_attachment_size(config:get("couchdb", "max_attachment_size", "infinity")).
+
+max_attachment_size(MaxAttSizeConfig) ->
+    case MaxAttSizeConfig of
         "infinity" ->
-            infinity;
+            ?ATT_SIZE_LIMIT;
+        MaxAttSize when is_list(MaxAttSize) ->
+            try list_to_integer(MaxAttSize) of
+                Result -> min(Result, ?ATT_SIZE_LIMIT)
+            catch _:_ ->
+                couch_log:error("invalid config value for max attachment size: ~p ", [MaxAttSize]),
+                throw(internal_server_error)
+            end;
+        MaxAttSize when is_integer(MaxAttSize) ->
```

Comment:
`config:get/3` always return a string. Which means the `when is_integer(MaxAttSize) -> ` and `MaxAttSize ->` clauses are not needed.